ts/indicators/ichimoku-kinko-hyo.src.ts

```diff
@@ -0,0 +1,1009 @@
+                });
+
+                // Now save line
+                indicator.graphsenkouSpan = indicator.graph;
+            }
+
+            // Clean temporary properties:
+            delete indicator.nextPoints;
+            delete indicator.fillGraph;
+
+            // Restore options and draw the Tenkan line:
+            indicator.points = mainLinePoints;
+            indicator.options = mainLineOptions;
+            indicator.graph = mainLinePath;
+        },
+        getGraphPath: function (
+            this: Highcharts.IKHIndicator,
+            points: Array<Highcharts.Point>
+        ): Highcharts.SVGPathArray {
+            var indicator = this,
+                path: Highcharts.SVGPathArray = [],
+                spanA: Highcharts.SVGPathArray,
+                fillArray: Highcharts.SVGPathArray = [],
+                spanAarr: Highcharts.SVGPathArray = [];
+
+            points = points || this.points;
+
+
+            // Render Senkou Span
+            if (indicator.fillGraph && indicator.nextPoints) {
+
+                spanA = SMA.prototype.getGraphPath.call(
+                    indicator,
+                    // Reverse points, so Senkou Span A will start from the end:
+                    indicator.nextPoints
+                );
+
+                spanA[0] = 'L';
+
+                path = SMA.prototype.getGraphPath.call(
+                    indicator,
+                    points
+                );
+
+                spanAarr = spanA.slice(0, path.length);
+
+                for (var i = (spanAarr.length - 1); i > 0; i -= 3) {
+                    fillArray.push(
+                        spanAarr[i - 2],
+                        spanAarr[i - 1],
+                        spanAarr[i]
+                    );
+                }
+                path = path.concat(fillArray);
+
+            } else {
+                path = SMA.prototype.getGraphPath.apply(indicator, arguments);
+            }
+
+            return path;
+        },
+        getValues: function (
+            series: Highcharts.Series,
+            params: Highcharts.IKHIndicatorParamsOptions
+        ): (boolean|Highcharts.IndicatorUndefinableValuesObject) {
+
+            var period: number = (params.period as any),
+                periodTenkan: number = (params.periodTenkan as any),
+                periodSenkouSpanB: number = (params.periodSenkouSpanB as any),
+                xVal: Array<number> = (series.xData as any),
+                yVal: Array<Array<number>> = (series.yData as any),
+                xAxis: Highcharts.Axis = series.xAxis,
+                yValLen: number = (yVal && yVal.length) || 0,
+                closestPointRange: number = (
+                    getClosestPointRange(xAxis) as any
+                ),
+                IKH: Array<Array<(number|undefined)>> = [],
+                xData: Array<number> = [],
+                dateStart: number,
+                date: (number|undefined),
```

Comment:
Try `date` with `number` only, as it is mandatory set in the for loop with the xVals. If this is working, you can remove all `date as any`.

Reply:
Unfortunately then the compiler reports `Variable 'date' is used before being assigned.` in the for-loop that is just before the return statement.

Reply:
Okay. Therefor the type is good as it is. 👍